components/claim/create-reward-claim.tsx

```diff
@@ -143,8 +158,45 @@ const CreateRewardClaim = () => {
     const startTimeStamp = claimInput.fromDate?.getTime();
     const endTimeStamp = claimInput.toDate?.getTime();
     const totalAmountOfTokens = claimInput.amount;
+    const currentDateTimeStamp = new Date().getTime();
+
+    // This is messy. I will turn this into a util function or something.
 
-    if (!startTimeStamp || !endTimeStamp || !totalAmountOfTokens) {
+    if (!startTimeStamp) {
+      showCustomErrorToast("Start timestamp missing.", width);
+      setLoading(false);
+      return;
+    } else if (!endTimeStamp) {
+      showCustomErrorToast("End timestamp missing.", width);
+      setLoading(false);
+      return;
+    } else if (endTimeStamp < startTimeStamp) {
+      showCustomErrorToast("End date before start date", width);
+      setLoading(false);
+      return;
+    } else if (
+      startTimeStamp < currentDateTimeStamp ||
```

Comment:
we'll have to check with @saksijas about this since they could possibly select past dates as start and end timestamps

Reply:
I understand past dates for start times but why for end timestamps? That doesn't really make sense to me? ':D